tests/common/value_helper.h

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2020-2022 The Khronos Group Inc.
+//  Copyright:	(c) 2017 by Codeplay Software LTD. All Rights Reserved.
+//
+//  This file contains helper functions for modifying and comparing values,
+//  arrays, and objects that implements operator[]
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_COMMON_VALUE_HELPER_H
+#define __SYCLCTS_TESTS_COMMON_VALUE_HELPER_H
+#include "../../util/type_traits.h"
+
+#include <cassert>
+
+namespace value_helper {
```

Comment:
IMO `value_helper` is too nondescript. I realize that (some of) these helpers also work on scalars, but the only reason to use them would be in a generic context where values can be array-like, right? So maybe something like `array_value_helper` or `vector_value_helper` would be clearer?

Reply:
I agree, that `value_halper` doesn't provide enough description, but if we call it `vector_value_helper` it will not be clear that functions can also work with scalar values too. As this is not a class, maybe we can call namespace `value_operations`? It will be more clearer and accurate because the namespace actually contains functions for assigning and comparing operations for scalar and vector types.

Reply:
Yes `value_operations` sounds better I think. Another idea could be something like `element_wise_operations` (where a scalar could be considered a vector with a single element). But either is fine with me.

Reply:
Changed namespace to `value_operations`